src/internal/template/template.go

```diff
@@ -45,22 +45,22 @@ func Generate() Values {
 	generated.seedRegistry = config.GetSeedRegistry()
 	generated.registry = config.GetRegistry()
 
-	generated.secret.registryPush = config.GetSecret(config.StateRegistryPush)
-	generated.secret.registryPull = config.GetSecret(config.StateRegistryPull)
-	generated.secret.registrySecret = config.GetSecret(config.StateRegistrySecret)
+	generated.secret.registryPush = config.GetContainerRegistryInfo().PushPassword
+	generated.secret.registryPull = config.GetContainerRegistryInfo().PullPassword
+	generated.secret.registrySecret = utils.RandomString(48)
```

Comment:
I do think if we are exposing this secret with templating we need to preserve it in the state too.  Otherwise things might get weird on a registry upgrade + template as you'd be introducing a different shared secret with the update.